account_setup.py

```diff
@@ -0,0 +1,578 @@
+        :param prompt: str
+        :param deny_error: Exception
+        :param unrecognized_input_error: Exception
+        :return None if prompt successful, else raises error
+    """
+    confirmation = input(prompt).lower()
+    for i in range(3):
+        if confirmation == "n":
+            raise deny_error
+        elif confirmation == "y":
+            break
+        elif confirmation != "y" and i == 2:
+            raise unrecognized_input_error
+        confirmation = input("Please input 'y' or 'n': ").lower()
+
+
+def prompt_with_default(key, value):
+    user_value = input("{0} [{1}]: ".format(key, value))
+    if user_value != "":
+        return user_value
+    else:
+        return value
+
+
+def prompt_tenant_selection(tenant_ids):
+    print("Multiple tenants detected. Please input the ID of the tenant you wish to use.")
+    print("Tenants:", ", ".join(tenant_ids))
+    given_tenant_id = input("Please input the ID of the tenant you wish to use: ")
+    for i in range(3):
+        if given_tenant_id in tenant_ids:
+            return given_tenant_id
+        if i != 2:
+            given_tenant_id = input("Input not recognized, please try again: ")
+    raise AccountSetupError("Tenant selection not recognized after 3 attempts.")
+
+
+
+class Spinner:
+    busy = False
+    delay = 0.1
+
+    @staticmethod
+    def spinning_cursor():
+        while 1:
+            for cursor in '|/-\\': yield cursor
+
+    def __init__(self, delay=None):
+        self.spinner_generator = self.spinning_cursor()
+        if delay and float(delay): self.delay = delay
+
+    def __enter__(self):
+        return self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return self.stop()
+
+    def spinner_task(self):
+        while self.busy:
+            sys.stdout.write(next(self.spinner_generator))
+            sys.stdout.flush()
+            time.sleep(self.delay)
+            sys.stdout.write('\b')
+            sys.stdout.flush()
+
+    def start(self):
+        self.busy = True
+        threading.Thread(target=self.spinner_task, daemon=True).start()
+
+    def stop(self):
+        self.busy = False
+        time.sleep(self.delay)
+
+
+
+if __name__ == "__main__":
+    print("\nGetting credentials.")
+    # get credentials and tenant_id
+    creds, subscription_id = credentials.get_azure_cli_credentials()
+    subscription_client = SubscriptionClient(creds)
+    tenant_ids = [tenant.id for tenant in subscription_client.tenants.list()]
+    if len(tenant_ids) != 1:
+        tenant_id = prompt_tenant_selection(tenant_ids)
+    else:
+        tenant_id = tenant_ids[0]
+
+    if len(sys.argv) > 1 and sys.argv[1] == "deleteresourcegroup":
+        resource_group = input("Resource Group Name: ")
+        with Spinner():
+          delete_resource_group(creds, subscription_id, resource_group)
+        print("Deleted resource group.")
+        sys.exit()
+   
+    if len(sys.argv) > 1 and sys.argv[1] == "serviceprincipal":
```

Comment:
Kind of strange to set authentication to "serviceprincipal" when the default is "1". Probably worth updating the default to be "sharedkey" instead of "1".

Reply:
replace "1" with "sharedkey"